src/main/kotlin/app/revanced/patches/unifiprotect/localdevice/directconnections/DirectConnectionsPatch.kt

```diff
@@ -0,0 +1,162 @@
+package app.revanced.patches.unifiprotect.localdevice.directconnections
+
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructionsWithLabels
+import app.revanced.patcher.extensions.or
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchException
+import app.revanced.patcher.patch.annotation.CompatiblePackage
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patcher.patch.options.types.StringPatchOption.Companion.stringPatchOption
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.patches.unifiprotect.localdevice.directconnections.fingerprints.DirectConnectionsMethodFingerprint
+import com.android.tools.smali.dexlib2.AccessFlags
+import com.android.tools.smali.dexlib2.builder.MutableMethodImplementation
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethod
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethodParameter
+
+@Patch(
+    name = "Enable direct connections",
+    compatiblePackages = [CompatiblePackage("com.ubnt.unifi.protect")],
+)
+@Suppress("unused")
+object DirectConnectionsPatch : BytecodePatch(
+    setOf(
+        DirectConnectionsMethodFingerprint
+    )
+) {
+    private var model by stringPatchOption(
+        "Model",
+        "UCK-G2-PLUS",
+        "The model to use for the direct connection device"
+    )
+
+    private var IP by stringPatchOption(
+        "IP",
+        "",
+        "The IP to use for the direct connection device"
+    )
+
+    private var MAC by stringPatchOption(
+        "MAC",
+        null,
+        "The MAC to use for the direct connection device"
+    )
+
+    override fun execute(context: BytecodeContext) {
+
+        val result = DirectConnectionsMethodFingerprint.result ?: throw PatchException("Could not find method to patch")
```

Comment:
Throw using the extension function: `throw DirectConnectionsMethodFingerprint.exception`